### PR TITLE
Images grid carousel #38

### LINKED ISSUE #38: Images grid carousel
Extension of #24. Upon clicking an image in the images grid, a [modal window will appear with a carousel](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=3514-6008&mode=dev).

### PR DESCRIPTION
**Pull request by TomaszDziezykNetcentric**
Fix #38 

Test URLs:
- Before: https://develop--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/images-grid-with-modal
- After: https://38-images-grid-carousel--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/images-grid-with-modal

### CODE REVIEW

**Review by SantiagoHomps-NC — APPROVED** [review]

File: blocks/v2-images-grid/v2-images-grid.js [hunk 1]
```diff
@@ -1,8 +1,117 @@
...
+    observer.observe(carouselItem);
+  });
+
+  const itemsWrapper = createElement('div', { classes: `${blockClassName}__carousel-items-wrapper` });
+  const wrapper = createElement('div', { classes: `${blockClassName}__modal-content` });
+
+  itemsWrapper.innerHTML = `
+    <div class="${blockClassName}__modal-arrows-wrapper">
+      <button class="${blockClassName}__modal-arrow">
+        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
```
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply blocks/v2-images-grid/v2-images-grid.js]: Updated

File: blocks/v2-images-grid/v2-images-grid.js [hunk 2]
```diff
@@ -1,8 +1,117 @@
...
+
+    observer.observe(carouselItem);
+  });
+
+  const itemsWrapper = createElement('div', { classes: `${blockClassName}__carousel-items-wrapper` });
+  const wrapper = createElement('div', { classes: `${blockClassName}__modal-content` });
+
+  itemsWrapper.innerHTML = `
+    <div class="${blockClassName}__modal-arrows-wrapper">
+      <button class="${blockClassName}__modal-arrow">
```
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply blocks/v2-images-grid/v2-images-grid.js]: Updated

File: blocks/v2-images-grid/v2-images-grid.css [hunk 3]
```diff
@@ -48,11 +65,145 @@
...
+}
+
+.v2-images-grid__modal-arrows-wrapper
+  .v2-images-grid__modal-arrow:last-of-type {
+  top: 50%;
+  right: 0;
+  position: absolute;
+}
+
+.v2-images-grid__carousel-preview-item button {
```
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply blocks/v2-images-grid/v2-images-grid.css:205]: Updated

File: common/modal/modal-component.css [hunk 4]
```diff
@@ -72,24 +73,44 @@
...
 
+@media (min-width: 768px) {
+  .redesign-v2 .modal-content {
+    max-width: 1040px;
+    height: auto;
+    position: relative;
+    aspect-ratio: unset;
+  }
+
+  .redesign-v2 .modal-background button.modal-close-button {
```
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply common/modal/modal-component.css:108]: Updated

File: blocks/v2-images-grid/v2-images-grid.js [hunk 5]
```diff
@@ -1,8 +1,117 @@
...
+        <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+          <path fill="var(--color-icon, #000)" d="M3 11C2.44772 11 2 11.4477 2 12C2 12.5523 2.44772 13 3 13L3 11ZM21.7071 12.7071C22.0976 12.3166 22.0976 11.6834 21.7071 11.2929L15.3431 4.92893C14.9526 4.53841 14.3195 4.53841 13.9289 4.92893C13.5384 5.31946 13.5384 5.95262 13.9289 6.34315L19.5858 12L13.9289 17.6569C13.5384 18.0474 13.5384 18.6805 13.9289 19.0711C14.3195 19.4616 14.9526 19.4616 15.3431 19.0711L21.7071 12.7071ZM3 13L21 13V11L3 11L3 13Z" />
+        </svg>
+      </button>
+    </div>
+  `;
+  itemsWrapper.append(carouselItemsList);
+
+  [...itemsWrapper.querySelectorAll('button')].forEach((el, elIndex) => {
```
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply blocks/v2-images-grid/v2-images-grid.js:110]: Done

File: blocks/v2-images-grid/v2-images-grid.css [hunk 6]
```diff
@@ -48,11 +65,163 @@
...
+
+.v2-images-grid__carousel-items-list::-webkit-scrollbar,
+.v2-images-grid__carousel-preview-list::-webkit-scrollbar {
+  display: none;
+}
+
+.v2-images-grid__carousel-preview-list {
+  gap: 10px;
+  display: flex;
+  list-style: none;
```
Comment (Lakshmishri) [inline blocks/v2-images-grid/v2-images-grid.css:103]: Some of the styles are repeated above at rule 82. So these can be removed (display, list-style, margin, padding

**Review by Lakshmishri — CHANGES_REQUESTED** [review]

**Review by Lakshmishri — CHANGES_REQUESTED** [review]
Screenshot 2023-11-15 at 20 54 14
I think the space should be less . @cogniSyb  - wdyt ?

File: blocks/v2-images-grid/v2-images-grid.css (hunk 6 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-images-grid/v2-images-grid.css]: done

**Comment by TomaszDziezykNetcentric** [comment]
> I noticed some odd scroll behaviour when navigating in the preview list. It’s also present on desktop, but due to the small viewport on mobile, it’s more distracting: https://github.com/Netcentric/vg-macktrucks-com-rd/assets/133873665/52bcc85f-4c01-4dce-aa8d-71ff51042180

Fixed

**Comment by taimurCognizant** [comment]
@cogniSyb @TomaszDziezykNetcentric @Lakshmishri 
I also have a generic comment on modal open/close transition. It is very slow, taking 1 second. And since the transition is applied on the height property I can see the model expanding. Moreover, Adding transition to height property is heavy on performance and on slower device the animation would be choppy.
Not sure if that is expected behavior or not. If we have freedom to apply any kind of transition I would suggest to transition only the `opacity` property

**Comment by TomaszDziezykNetcentric** [comment]
> Screenshot 2023-11-15 at 20 54 14 I think the space should be less . @cogniSyb - wdyt ?

I was testing it on smaller devices... I agree that it looks weird now :( 
FIxed

**Comment by TomaszDziezykNetcentric** [comment]
> @cogniSyb @TomaszDziezykNetcentric @Lakshmishri I also have a generic comment on modal open/close transition. It is very slow, taking 1 second. And since the transition is applied on the `height` property I can see the model expanding. Moreover, Adding transition to `height` property is heavy on performance and on slower device the animation would be choppy. Not sure if that is expected behavior or not. If we have freedom to apply any kind of transition I would suggest to transition only the `opacity` property

We can update the animation, but I would like to do this in the scope of the other issue :) 
The change will be very easy, but because we are using this modal at other paces that are out of the scope of the redesign it would be better to test it in a separate issue scope.

**Review by Lakshmishri — APPROVED** [review]

File: blocks/v2-images-grid/v2-images-grid.css [hunk 7]
```diff
@@ -48,11 +65,161 @@
...
+}
+
+.v2-images-grid__carousel-item picture {
+  display: flex;
+}
+
+.v2-images-grid__modal-content .v2-images-grid__carousel-item img {
+  aspect-ratio: 16/9;
+  width: 100%;
+}
```
[the remark this answers is not in the document]
Comment (taimurCognizant) [reply blocks/v2-images-grid/v2-images-grid.css:139]: @TomaszDziezykNetcentric  can you see the comment?

**Review by taimurCognizant — CHANGES_REQUESTED** [review]

File: blocks/v2-images-grid/v2-images-grid.css (hunk 7 above)
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply blocks/v2-images-grid/v2-images-grid.css:140]: done

**Review by taimurCognizant — APPROVED** [review]

**Pull request merged**